### PR TITLE
🎓 Descriptions for Entries

### PR DESCRIPTION
**Pull request by Gilad-Gur-Andelman**

### CODE REVIEW

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/2dbeafec-d009-4b7d-9d11-7d814da5fa21/artifacts/0/out/make/Loadmill-3.1.27-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/2dbeafec-d009-4b7d-9d11-7d814da5fa21/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.27.zip)

File: src/main-process/proxy/request-description.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { LoadmillRequest } from './test-actions/transform';
+
+export const assignRequestDescriptions = (entries: ProxyEntry[], requests: LoadmillRequest[]): void => {
+  entries.forEach(entry => assignRequestDescription(entry, requests));
+};
+
+const assignRequestDescription = (entry: ProxyEntry, requests: LoadmillRequest[]): void => {
+  const request = requests.find(request => request.id === entry.id);
+  if (request) {
+    entry.request.description = request.description;
```
Comment (onniaa) [inline src/main-process/proxy/request-description.ts:12]: `|| getDefaultRequestDescription` to be on the safe side if the request returns without description?

File: src/main-process/proxy/test-actions/analyze.ts [hunk 2]
```diff
@@ -60,6 +61,7 @@ const pollTransformStatus = async (token: string): Promise<void> => {
         const requests = details.conf.requests;
         const entries = getEntries();
         markIrrelevantRequests(entries, requests);
+        assignRequestDescriptions(entries, requests);
```
Comment (onniaa) [inline src/main-process/proxy/test-actions/analyze.ts:64]: maybe this and `markIrrelevantRequests` can be in the same function that process/prepare entries? This way we don't iterate twice on the entries and also on future operations we won't have to add another function here.

File: src/main-process/proxy/test-actions/transform.ts [hunk 3]
```diff
@@ -81,7 +81,14 @@ export const getTransformResult = async (
 };
 
 export type Extraction = { [parameter: string]: string | object };
-export type LoadmillRequest = { extract: Extraction[]; id: string; irrelevant?: boolean; method: string; };
+export type LoadmillRequest = {
+  description: string;
```
Comment (onniaa) [inline src/main-process/proxy/test-actions/transform.ts:85]: can `description` be `undefined`?

Comment (Gilad-Gur-Andelman) [reply src/main-process/proxy/test-actions/transform.ts:85]: Yes. Although this is a rare case, we will handle it.

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/c3636037-b03e-4148-887a-f99ec116be37/artifacts/0/out/make/Loadmill-3.1.27-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/c3636037-b03e-4148-887a-f99ec116be37/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.1.27.zip)

**Comment by Gilad-Gur-Andelman** [comment]
## 🏗 Build Artifacts are available here👇 

### 1. [🍏 .dmg](https://output.circle-artifacts.com/output/job/edae807b-eec9-4141-b5c8-570a12b2c26d/artifacts/0/out/make/Loadmill-3.2.0-x64.dmg)

### 2. [🤐 .zip](https://output.circle-artifacts.com/output/job/edae807b-eec9-4141-b5c8-570a12b2c26d/artifacts/0/out/make/zip/darwin/x64/Loadmill-darwin-x64-3.2.0.zip)

**Review by onniaa — APPROVED** [review]

**Pull request merged**